### PR TITLE
Fixed check for AdditionalFiles

### PR DESCRIPTION
**Pull request by duncanp-sonar**

### CODE REVIEW

File: src/Integration/Binding/CSharpVBProjectBinder.cs [hunk 1]
```diff
@@ -156,10 +155,17 @@ private RuleSet FindDeclarationRuleSet(Project project, RuleSetDeclaration decla
             return ruleSetSerializer.LoadRuleSet(ruleSetFilePath);
         }
 
-        private bool ProjectHasAdditionalFile(Project project, string additionalFilePath)
+        internal /* for testing */ static bool IsAdditionalFileReferencedCorrectly(IProjectSystemHelper projectSystemHelper, Project project, string additionalFilePath)
```
Comment (duncanp-sonar) [inline src/Integration/Binding/CSharpVBProjectBinder.cs:158]: `static` to make it easy to test in isolation.

**Comment by duncanp-sonar** [comment]
I suggest looking at the commits separately.
The first is a refactoring so that `GetFileInProject` and `IsFileInProject` call a common helper function.

File: src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs [hunk 2]
```diff
@@ -167,6 +168,61 @@ public void ProjectSystemHelper_IsFileInProject_DifferentCase()
...
+
+            // 2. Null file name
+            act = () => this.testSubject.FindFileInProject(new ProjectMock("project"), null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("file");
+
+            // 3. Empty file name
+            act = () => this.testSubject.FindFileInProject(new ProjectMock("project"), "");
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("file");
+
+            // 4. Null file name
```
Comment (rita-g-sonarsource) [inline src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs:195]: The comment is misleading, maybe "whitespace file name"?

File: src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs [hunk 3]
```diff
@@ -93,16 +94,16 @@ public void ProjectSystemHelper_GetSolutionProjects_ReturnsOnlyKnownLanguages()
         {
             // Arrange
             ProjectMock csProject = this.solutionMock.AddOrGetProject("c#");
-            csProject.SetProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID.VSHPROPID_ExtObject, csProject);
+            csProject.SetExtObjProperty(VSConstants.VSITEMID_ROOT, csProject);
```
Comment (rita-g-sonarsource) [inline src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs:97]: Is SetExtObjProperty used for anything else?
We can move it to ProjectMock's ctor, to always set VSITEMID_ROOT to point to `this`

File: src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs:195]: I copied and pasted this from the `AddFileToProject` test. I'll fix both.

**Pull request merged**